### DISCUSSION TITLE
How to set clock speed?

### DISCUSSION THREAD

**Discussion by SquashSquishSquaw** [SpenceKonde/megaTinyCore#460] [Q&A] [answered]
Hi,

sorry for this newbie question:

In Arduino 1.8.15 with megaTinyCore, if I set "Tools -> Clock (burn bootloader not req'd)" to 1 MHz, is that all I need to do to change clock speed? Or do I have to write code instructions as well, for example in the setup code?

And when my ATtiny 1624 finally runs at 1 MHz, will it automatically run with only 1.8 V, or do I need to set this specifically, somewhere?

Many thanks
Jenny

**Comment by grandaspanna** [comment, marked as answer]
In general, yes, that's all you have to do. Things should work without any specific code of your own. Test with the blink example.

Your 1624 should be stable with any supply voltage from 1.8V to 5.5V @ 1MHz. According to datasheet, can run at 5MHz from 1.8V.

**Reply by SquashSquishSquaw** [reply]
Many thanks grandaspanna, that's a good idea!

I can test whether it works with 1.8 V or not, but measuring clock speed I'm not sure how to do. I'd like to have it run on 1 MHz (or even less if possible), in order to save power.

However, if it doesn't run on 1.8 V any more after I switch back to 10 MHz, then I guess I can safely assume that the switching without further code works.

**Reply by SpenceKonde** [reply]
To check that it is running at the speed 6ou thst you selected (and hence what it is compiled for), blink will blink at the wrong speed if it is compiled for different F_CPU).

To verify that they arent both wrong but the same, take over tca0, put back into single mode, divide clock ny 1024 (maximum) so TCA tick = approx 1ms, set TCA0.PER to 1999 (or 1953 9f you feels need to correct for 1024 vs 1000) and one of the cmp registers for 999 (or 976) and at 1 mhz, 5oy should get 1 sec blink. If you have a scope, of course, dispense with the 1024 prescale, 999 PER 499 CMP for 1 khz output, I do that all th we time

Phone sent, set for typos

**Reply by SpenceKonde** [reply]
And yeah, the init() function which is called before setup calls sub functions to make pwm, adc,   millis timing and that kind of shit work. First one it calls sets up system clock

**Reply by SpenceKonde** [reply]
And when uploading via updi the 20 vs 16 derived clock fuse is set (note that this 8s not the case 2ith opriboot board def, though it will be possible when tuning goes in in 1.4.0, any at now

**Reply by SquashSquishSquaw** [reply]
Many thanks Spence, for all the help, much appreciated!

I'll give my best trying to learn how to do these things. :-D

Best regards, have a nice week
Jenny

**Reply by coderatchance** [reply]
I am on 1.8.13 and recent megaTinyCore with the 1626 and somehow I cannot get the print out running on 115200 , did burned the bootloader and the code is telling me I am on 20Mhz, however, the print out does not work, am I really on 20Mhz.

Any ideas where I am getting it wrong?
